XenModel/Utils/Helpers.cs

```diff
@@ -352,6 +352,25 @@ public static bool ElyOrGreater(Host host)
                 platform_version != null && productVersionCompare(platform_version, "2.1.1") >= 0;
         }
 
+        public static bool HavanaOrGreaterLTSR71(IXenConnection conn)
+        {
+            return conn == null || HavanaOrGreaterLTSR71(Helpers.GetMaster(conn));
+        }
+
+        /// As Havana platform version is same with Ely and Honolulu, so use product version here
+        /// <param name="host">May be null, in which case true is returned.</param>
+        public static bool HavanaOrGreaterLTSR71(Host host)
+        {
+            if (host == null)
+                return true;
+             string product_version = HostProductVersion(host);
+            return
+                product_version != null && 
+                product_version.StartsWith("[BRANDING_XC_PRODUCT_7_1_VERSION]") && 
+                productVersionCompare(product_version, "BRANDING_XC_PRODUCT_7_1_2_VERSION") >= 0 && 
+                productVersionCompare(product_version, "BRANDING_XC_PRODUCT_7_2_VERSION") < 0;

```

Comment:
I don't think that `productVersionCompare(product_version, "BRANDING_XC_PRODUCT_7_2_VERSION") < 0;` is needed, as `product_version.StartsWith("[BRANDING_XC_PRODUCT_7_1_VERSION]") ` will exclude anything greater than 7.2.
Alternatively, we can say `ElyOrGreater && !FalconOrGreater && productVersionCompare(product_version, "BRANDING_XC_PRODUCT_7_1_2_VERSION") >= 0`

Reply:
I'm afraid this won't work for icenter.